Exception occurred when execute set name "utf8" in ShardingSphere-Proxy with MySQL client

## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?
master

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy + MySQL

### Expected behavior
`set names "utf8";` execute successfully

### Actual behavior

Exception occurred :

```
[ERROR] 2022-07-05 12:02:52.770 [ShardingSphere-Command-0] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
java.nio.charset.IllegalCharsetNameException: "utf8"
	at java.nio.charset.Charset.checkName(Charset.java:315)
	at java.nio.charset.Charset.lookup2(Charset.java:484)
	at java.nio.charset.Charset.lookup(Charset.java:464)
	at java.nio.charset.Charset.forName(Charset.java:528)
	at org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.MySQLSetCharsetExecutor.parseCharset(MySQLSetCharsetExecutor.java:63)
	at org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.MySQLSetCharsetExecutor.execute(MySQLSetCharsetExecutor.java:47)
	at org.apache.shardingsphere.proxy.backend.text.encoding.DatabaseSetCharsetBackendHandler.execute(DatabaseSetCharsetBackendHandler.java:43)
	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.execute(MySQLComQueryPacketExecutor.java:99)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:108)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:78)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

```

### Reason analyze (If you can)

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.

execute `set names "utf8";` in ShardingSphere-Proxy

### Example codes for reproduce this issue (such as a github link).

--- a/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/MySQLSetCharsetExecutor.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/MySQLSetCharsetExecutor.java
@@ -27,6 +27,7 @@ import org.apache.shardingsphere.sql.parser.sql.common.statement.dal.SetStatemen
 
 import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
+import java.nio.charset.UnsupportedCharsetException;
 import java.sql.SQLException;
 import java.util.Locale;
 
@@ -50,7 +51,7 @@ public final class MySQLSetCharsetExecutor implements DatabaseSetCharsetExecutor
     }
     
     private String formatValue(final String value) {
-        return value.startsWith("'") && value.endsWith("'") ? value.substring(1, value.length() - 1) : value.trim();
+        return value.startsWith("'") && value.endsWith("'") || value.startsWith("\"") && value.endsWith("\"") ? value.substring(1, value.length() - 1) : value.trim();
     }
     
     private Charset parseCharset(final String value) {
@@ -60,7 +61,13 @@ public final class MySQLSetCharsetExecutor implements DatabaseSetCharsetExecutor
             case "utf8mb4":
                 return StandardCharsets.UTF_8;
             default:
-                return Charset.forName(value);
+                try {
+                    return Charset.forName(value);
+                    // CHECKSTYLE:ON
+                } catch (Exception ex) {
+                    // CHECKSTYLE:OFF
+                    throw new UnsupportedCharsetException(value.toLowerCase());
+                }
         }
     }
     

--- a/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/MySQLSetCharsetExecutor.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/MySQLSetCharsetExecutor.java
@@ -63,9 +63,9 @@ public final class MySQLSetCharsetExecutor implements DatabaseSetCharsetExecutor
             default:
                 try {
                     return Charset.forName(value);
-                    // CHECKSTYLE:ON
-                } catch (Exception ex) {
                     // CHECKSTYLE:OFF
+                } catch (Exception ex) {
+                    // CHECKSTYLE:ON
                     throw new UnsupportedCharsetException(value.toLowerCase());
                 }
         }

